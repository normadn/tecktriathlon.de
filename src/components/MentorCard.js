import React from "react";

export default ({
  headshot,
  name,
  title,
  companyLogo,
  borderColor,
  companyName,
  companyUrl,
  logoWidth = "180px",
  logoHeight = "auto",
  logoMarginTop = "0px"
}) => (
  <div css={{ textAlign: "center", margin: "10px", marginTop: "56px" }}>
    <div
      css={{
        width: "180px",
        height: "180px",
        borderRadius: "50%",
        borderColor,
        marginLeft: "auto",
        marginRight: "auto",
        borderWidth: "3px",
        borderStyle: "solid",
        overflow: "hidden"
      }}
    >
      <img
        css={{ width: "190px", marginLeft: "-10px" }}
        src={headshot}
        alt={`${name} - ${title}`}
      />
    </div>

    <h2
      css={{
        marginTop: "24px",
        marginBottom: "8px",
        fontSize: "28px",
        fontWeight: "600"
      }}
    >
      {name}
    </h2>
    <p>{title}</p>
    <br />
    <a href={companyUrl} target="_blank" rel="external noopener">
      <img
        css={{
          width: logoWidth,
          height: logoHeight,
          marginBottom: "32px",
          marginTop: logoMarginTop
        }}
        src={companyLogo}
        alt={companyName}
      />
    </a>
  </div>
);
