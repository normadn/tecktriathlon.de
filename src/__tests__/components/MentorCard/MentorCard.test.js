import React from "react";
import MentorCard from "../../../components/MentorCard";
import { shallow } from "enzyme";

const mentor = {
  name: "Eric Braun",
  title: "Machine Learning Engineer",
  headshot: "eric.png",
  companyLogo: "qualcomm.png",
  companyName: "Qualcomm",
  companyUrl: "https://www.qualcomm.com/"
};

describe("MentorCard", () => {
  it("renders correctly", () => {
    expect(shallow(<MentorCard {...mentor} />)).toMatchSnapshot();
  });
});
