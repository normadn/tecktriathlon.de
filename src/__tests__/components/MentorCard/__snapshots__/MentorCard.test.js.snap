// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorCard renders correctly 1`] = `
<div
  css={
    Object {
      "margin": "10px",
      "marginTop": "56px",
      "textAlign": "center",
    }
  }
>
  <div
    css={
      Object {
        "borderColor": undefined,
        "borderRadius": "50%",
        "borderStyle": "solid",
        "borderWidth": "3px",
        "height": "180px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "overflow": "hidden",
        "width": "180px",
      }
    }
  >
    <img
      alt="Eric Braun - Machine Learning Engineer"
      css={
        Object {
          "marginLeft": "-10px",
          "width": "190px",
        }
      }
      src="eric.png"
    />
  </div>
  <h2
    css={
      Object {
        "fontSize": "28px",
        "fontWeight": "600",
        "marginBottom": "8px",
        "marginTop": "24px",
      }
    }
  >
    Eric Braun
  </h2>
  <p>
    Machine Learning Engineer
  </p>
  <br />
  <a
    href="https://www.qualcomm.com/"
    rel="external noopener"
    target="_blank"
  >
    <img
      alt="Qualcomm"
      css={
        Object {
          "height": "auto",
          "marginBottom": "32px",
          "marginTop": "0px",
          "width": "180px",
        }
      }
      src="qualcomm.png"
    />
  </a>
</div>
`;
